runtime/src/chpl-file-utils.c

```diff
@@ -113,6 +117,78 @@ qioerr chpl_fs_exists(int* ret, const char* name) {
   return err;
 }
 
+int chpl_fs_get_cuid(void) {
+  return getuid();
+}
+
+/*
+  Quick implementation of strcpy() using qio_malloc(). If it already exists
+  I can easily replace this.
+*/
+char* copy_string(const char* input);
+char* copy_string(const char* input) {
+  char* result = NULL;
+  size_t len = strlen(input);
+
+  result = qio_malloc(len + 1);
+
+  if (result == NULL) { return NULL; }
+
+  result[len] = 0;
+  strncpy(result, input, len);
+
+  return result;
+}
+
+/* Common code shared by "chpl_fs_get_home()" and "chpl_fs_get_home_uid()". */
+static qioerr
+copy_passwd_login_dir(const char** outs, int* outf, struct passwd* pw) {
+  char* result_str = NULL;
+  int result_found = 0;
+
+  // If we've found the given user, attempt to copy their login dir.
+  if (pw != NULL) {
+    result_str = copy_string(pw->pw_dir);
+    if (result_str == NULL) { return qio_mkerror_errno(); }
```

Comment:
If this function is getting the errno from copy_string, well, don't do that, copy_string shouldn't set errno.
If it's trying to propagate error from another (previous) call, I think that would be more clearly handled at the call site.